### ISSUE TITLE
[AKS] `test_update_default_mc_profile` fails with "does not have the attribute '_get_rg_location'"

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-cli#20532]
**Describe the bug**

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1216795&view=logs&j=771f6cdf-8d98-5d9b-2fb9-d3a6e9554788&t=0a182f1d-9946-580d-467e-4d4ebd6af32d&l=3913

```
2021-11-29T04:30:28.3858645Z >       with patch(
2021-11-29T04:30:28.3859163Z             "azure.cli.command_modules.acs.decorator._get_rg_location",
2021-11-29T04:30:28.3859742Z             return_value="test_location",
2021-11-29T04:30:28.3860226Z         ), patch(
2021-11-29T04:30:28.3860741Z             "azure.cli.command_modules.acs.decorator.Profile",
2021-11-29T04:30:28.3861535Z             return_value=mock_profile,
2021-11-29T04:30:28.3862300Z         ), patch(
2021-11-29T04:30:28.3863094Z             "azure.cli.command_modules.acs.decorator.AKSUpdateDecorator.check_raw_parameters",
2021-11-29T04:30:28.3863623Z             return_value=True,
2021-11-29T04:30:28.3864044Z         ), patch.object(
2021-11-29T04:30:28.3864494Z             self.client, "get", return_value=mock_existing_mc
2021-11-29T04:30:28.3865125Z         ):
2021-11-29T04:30:28.3865372Z 
2021-11-29T04:30:28.3866155Z src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py:7640: 
2021-11-29T04:30:28.3866849Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-29T04:30:28.3867462Z /opt/hostedtoolcache/Python/3.10.0/x64/lib/python3.10/unittest/mock.py:1438: in __enter__
2021-11-29T04:30:28.3867990Z     original, local = self.get_original()
2021-11-29T04:30:28.3868504Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-29T04:30:28.3868866Z 
2021-11-29T04:30:28.3869288Z self = <unittest.mock._patch object at 0x7fe8bc8599f0>
2021-11-29T04:30:28.3869606Z 
2021-11-29T04:30:28.3869968Z     def get_original(self):
2021-11-29T04:30:28.3870401Z         target = self.getter()
2021-11-29T04:30:28.3870822Z         name = self.attribute
2021-11-29T04:30:28.3871209Z     
2021-11-29T04:30:28.3871573Z         original = DEFAULT
2021-11-29T04:30:28.3871980Z         local = False
2021-11-29T04:30:28.3872362Z     
2021-11-29T04:30:28.3872727Z         try:
2021-11-29T04:30:28.3873136Z             original = target.__dict__[name]
2021-11-29T04:30:28.3873614Z         except (AttributeError, KeyError):
2021-11-29T04:30:28.3874099Z             original = getattr(target, name, DEFAULT)
2021-11-29T04:30:28.3874544Z         else:
2021-11-29T04:30:28.3874925Z             local = True
2021-11-29T04:30:28.3875611Z     
2021-11-29T04:30:28.3876170Z         if name in _builtins and isinstance(target, ModuleType):
2021-11-29T04:30:28.3876678Z             self.create = True
2021-11-29T04:30:28.3877075Z     
2021-11-29T04:30:28.3877512Z         if not self.create and original is DEFAULT:
2021-11-29T04:30:28.3877993Z >           raise AttributeError(
2021-11-29T04:30:28.3878458Z                 "%s does not have the attribute %r" % (target, name)
2021-11-29T04:30:28.3878916Z             )
2021-11-29T04:30:28.3879959Z E           AttributeError: <module 'azure.cli.command_modules.acs.decorator' from '/home/vsts/work/1/s/src/azure-cli/azure/cli/command_modules/acs/decorator.py'> does not have the attribute '_get_rg_location'
```

This is caused by #20453. +@FumingZhang

**Comment by FumingZhang** [comment]
The failed test case should have been fixed by [PR #20456](https://github.com/Azure/azure-cli/pull/20456). From the [CI log](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1217027&view=logs&j=771f6cdf-8d98-5d9b-2fb9-d3a6e9554788&t=0a182f1d-9946-580d-467e-4d4ebd6af32d) after this PR is merged , we could not find such error message anymore.

**Comment by yonzhan** [comment]
route to service team
